clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -36,6 +169,57 @@ public void getUri() throws URISyntaxException, IOException {
         Assert.assertNotNull(u);
     }
 
+    @Test
+    public void testExists_Exists() throws ApiException, IOException {
+        Path p = new Path("lakefs://repo/main/exis.ts");
+        when(objectsApi.statObject("repo", "main", "exis.ts")).
+            thenReturn(new ObjectStats());
+
+        Assert.assertTrue(fs.exists(p));
+    }
+    @Test
+    public void testExists_NotExists() throws ApiException, IOException {
+        Path p = new Path("lakefs://repo/main/doesNotExi.st");
+        when(objectsApi.statObject(any(), any(), any())).
+            thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "no such file"));
+        when(objectsApi.listObjects(any(), any(), any(), any(), any(), any())).
+            thenReturn(new ObjectStatsList().results(Collections.emptyList()));
+
+        Assert.assertFalse(fs.exists(p));
+    }
+
+    @Test
+    public void testCreate() throws ApiException, IOException {
+        String contents = "The quick brown fox jumps over the lazy dog.";
+        Path p = new Path("lakefs://repo/main/create.me");
+
+        when(stagingApi.getPhysicalAddress("repo", "main", "create.me")).
+            thenReturn(new StagingLocation().token("foo").physicalAddress(s3Url("/repo-base/create")));
+        // TODO(ariels): Verify call to lakeFS "link" -- or verify that lakeFSFS "open" works.
+
+        OutputStream out = fs.create(p);
+        out.write(contents.getBytes());
+        out.close();
+
+        // Write succeeded, verify physical file on S3.
+        S3Object ret = s3Client.getObject(new GetObjectRequest(s3Bucket, "/repo-base/create"));
+        InputStream in = ret.getObjectContent();
+        ByteArrayOutputStream actualBytes = new ByteArrayOutputStream();
+        {
+            int len;
+            byte[] buf = new byte[1024];
+            while ((len = in.read(buf)) > 0) {
+                actualBytes.write(buf, 0, len);
+            }
+        }
+        actualBytes.close();
+        String actual = actualBytes.toString();
```

Comment:
```suggestion
        String actual = IOUtils.toString(in);
```

Reply:
Thanks!  Your way also works and might be _just a bit_ more readable :-).